DiscreetORM.ts

```diff
@@ -134,6 +136,31 @@ export function WriteToDB(discreet_sql_io : DiscreetORMIO){
         return descriptor;
     }
 }
+/** Method decorator to be applied to functions that modify a databased backed object, in place. 
+ *  Executes the function, modifying the object. The object's existing DB record is deleted and 
+ * replaced with the modified result of the function. Associates DB objects by the secret field 'discreet_orm_id'. 
+ * 
+ * To be used on methods that modifies object in place (generall)
+ * Note: Will generally be used on dynamic methods of the class
+ */
+export function WriteModifiedToDB(discreet_sql_io : DiscreetORMIO){
+    return function (target: Object, propertyKey: string | symbol, descriptor: PropertyDescriptor) {
+        const original_function = descriptor.value;
+        
+        descriptor.value = function(... args: any[]) {
+            const binded_original_function = original_function.bind(this)
+            let result = binded_original_function(args)
+            let result_table_name = this.constructor.name
+            let reference_id = this['discreet_orm_id'];
+            let delete_row_template = 'DELETE FROM ?? WHERE ??;'
```

Comment:
I think we might want to consider handling this with an UPDATE instead of a DELETE-INSERT pair; this opens us up to the risk of a clientside crash causing record loss (unless we use a transaction) and seems to be unnecessarily complicated when we already have the unique ID of the row and can compute a diff between the original and changed values

Reply:
I used the same logic that was there earlier but yeah I do think an UPDATE would be better
@T40TDS thoughts?